iocore/hostdb/P_HostDBProcessor.h

```diff
@@ -190,6 +190,14 @@ extern RecRawStatBlock *hostdb_rsb;
 #define HOSTDB_DECREMENT_THREAD_DYN_STAT(_s, _t) RecIncrRawStatSum(hostdb_rsb, _t, (int)_s, -1);
 
 
+struct CmpConstBuffferCaseInsensitive {
+  bool operator()(ts::ConstBuffer a, ts::ConstBuffer b) { return ptr_len_casecmp(a._ptr, a._size, b._ptr, b._size) < 0;}
+};
+// A to hold a ref-counted map
+struct RefCountedHostsFileMap : public RefCountObj {
+  std::map<ts::ConstBuffer, IpAddr, CmpConstBuffferCaseInsensitive> hosts_file_map;
```

Comment:
You'll probably want `typedef std::map<ts::ConstBuffer, IpAddr, CmpConstBufferCaseInsensitive> MapType;' here. It's a handy typedef.